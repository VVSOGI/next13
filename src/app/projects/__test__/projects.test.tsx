import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Container from '../container'
import { useRouter } from 'next/navigation'
import ErrorComponents from '../error'

jest.mock('next/navigation', () => ({
    useRouter: jest.fn() // useRouter 훅 모의화
}))

const customFetch = jest.fn()

describe('Projects', () => {
    test('should render projects', async () => {
        customFetch.mockResolvedValueOnce([
            {
                id: '1',
                title: 'title'
            }
        ])
        const projects = await customFetch()

        ;(useRouter as jest.Mock).mockReturnValue({
            push: `/${projects.id}`
        })

        const { container } = render(<Container projects={projects} />)
        expect(container).toMatchSnapshot('render projects')
    })

    test('Should return error when fetch failed projects', async () => {
        const networkError = JSON.stringify({
            error: 'NetworkError',
            message: 'The server could not be reached. Please check your network connection.',
            statusCode: 500
        })
        customFetch.mockResolvedValueOnce(networkError)

        const error = new Error(await customFetch())

        const { container } = render(<ErrorComponents error={error} reset={() => {}} />)
        expect(container).toMatchSnapshot('error')
    })
})
