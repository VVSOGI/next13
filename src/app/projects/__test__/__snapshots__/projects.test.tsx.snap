// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Should return error when fetch failed projects: error 1`] = `
<div>
  <div>
    <p>
      500
    </p>
    <p>
      The server could not be reached. Please check your network connection.
    </p>
  </div>
</div>
`;

exports[`Projects should render projects: render projects 1`] = `
<div>
  <div>
    <div
      class="css-i78qt7"
    >
      <h1
        class="css-1lta3h2"
      >
        title
      </h1>
      <button
        class="css-1o3n102"
      >
        이동
      </button>
    </div>
  </div>
</div>
`;
